### DISCUSSION TITLE
[Feature Request] Implement Global Menu For KDE Plasma and Other DE

### DISCUSSION THREAD

**Discussion by sapphired18** [zen-browser/desktop#2737] [Ideas]
I really like the idea to have global menu in Plasma and Other DE.

https://forum.manjaro.org/t/howto-enabling-kde-global-menu-support-in-mozilla-firefox-and-thunderbird/7405?u=grahamperrin

https://github.com/rilian-la-te/vala-panel-appmenu/blob/master/subprojects/appmenu-gtk-module/README.md

**Comment by qguyot** [comment]
[Here](https://aur.archlinux.org/cgit/aur.git/tree/unity-menubar.patch?h=firefox-globalmenu) is the only version I could find of the patch that would allow to add the feature to Zen.

**Comment by gleyconJ** [comment]
Hello,

First, thank you for your work on Zen Browser! I really appreciate the focus on privacy and productivity.

I am a KDE Plasma 6 user and would love to see support for the Global Menu feature in Zen. Currently, Firefox integrates well with the KDE Global Menu, but Zen does not export its menu to the panel. Since Zen is based on Firefox, I hope it might be possible to implement this integration as well.

Global Menu support is important for a consistent desktop experience and is highly valued by many KDE users. If there are technical challenges or patches needed, I would be happy to help test or provide feedback.

Thank you for considering this request!

**Comment by arkgnan** [comment]
I hope this idea will be implemented in the next version in 2026
